tests/core/ast_transforms/test_basic_sanity.py

```diff
@@ -73,20 +73,23 @@ def test_child_transform_square_to_zero():
 
 
 def test_problematic_child_transform_with_faked_child():
-    # function with a faked name should be respected as an override
-    # TODO this test captures current problematic behaviour which should be fixed
+    # Because of the intricacies of the AST rewrite
+    # (global namespace per transformed function), it's hard
+    # to make sure we're resilient to function name faking.
+    #
+    # This is a known shortcoming of our approach
+    # - you're welcome to fix it!
```

Comment:
Changed the comment as agreed. I still believe it's possible to fix it - never say never!

Reply:
Should be good to merge now.